src/Util/TestResultCache.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Runner;
+
+class TestResultCache implements \Serializable
+{
+    /**
+     * @var string
+     */
+    public const DEFAULT_RESULT_CACHE_FILENAME = '.phpunit.result.cache';
+
+    /**
+     * Provide extra protection against incomplete or corrupt caches
+     *
+     * @var array<string, string>
+     */
+    private const ALLOWED_CACHE_TEST_STATUSES = [
+        BaseTestRunner::STATUS_SKIPPED,
+        BaseTestRunner::STATUS_INCOMPLETE,
+        BaseTestRunner::STATUS_FAILURE,
+        BaseTestRunner::STATUS_ERROR,
+        BaseTestRunner::STATUS_RISKY,
+        BaseTestRunner::STATUS_WARNING
+    ];
+
+    /**
+     * Alternative path and filename for result cache file
+     *
+     * @var null|string
```

Comment:
when could this be null, if it fallbacks to `DEFAULT_RESULT_CACHE_FILENAME` ?

Reply:
Good catch, cleaned up. In previous versions `null` was used to the default as a fallback in another part of the code.